csharp/src/Drivers/BigQuery/BigQueryConnection.cs

```diff
@@ -683,8 +691,40 @@ private XdbcDataType ToXdbcDataType(string type)
                     return XdbcDataType.XdbcDataType_XDBC_VARBINARY;
                 case "NUMERIC" or "DECIMAL" or "BIGNUMERIC" or "BIGDECIMAL":
                     return XdbcDataType.XdbcDataType_XDBC_NUMERIC;
-
                 default:
+
+                    int decimalMaxScale = 28;
```

Comment:
Is the "28" due to limitations of CLR decimal? How does this code relate to the use of SqlDecimal?

Reply:
Yes, that is the reason. In SqlDecimal.ToDecimal() if scale > 28 then an OverthrowException is thrown. This is the XDBC metadata that is matching the SqlDecimal type that would come back for the value. I added comments in line.